ios/keyman/Keyman/Keyman/KMWebBrowser/WebBrowserViewController.swift

```diff
@@ -0,0 +1,281 @@
+        }
+        let bgImg = image.resizableImage(
+          withCapInsets: UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0), resizingMode: .stretch)
+        navigationController?.navigationBar.setBackgroundImage(bgImg, for: UIBarMetrics.default)
+      }
+    } else {
+      let image: UIImage
+      if UIInterfaceOrientationIsPortrait(orientation) {
+        image = #imageLiteral(resourceName: "kmwb-navbar-Portrait.png")
+      } else {
+        image = #imageLiteral(resourceName: "kmwb-navbar-Landscape.png")
+      }
+      navigationController?.navigationBar.setBackgroundImage(image, for: UIBarMetrics.default)
+    }
+  }
+
+  override func didRotate(from fromInterfaceOrientation: UIInterfaceOrientation) {
+    navBarTopConstraint.constant = AppDelegate.statusBarHeight()
+  }
+
+  func loadAddress(_ sender: Any, event: UIEvent) {
+    guard let urlString = addressField?.text, var url = URL(string: urlString) else {
+      NSLog("Attempting to load invalid URL: %@", addressField?.text ?? "nil")
+      return
+    }
+    if url.scheme == nil {
+      url = URL(string: "http://\(urlString)") ?? url
+    }
+    let request = URLRequest(url: url)
+    webView.loadRequest(request)
+  }
+
+  func updateAddress(_ request: URLRequest) {
+    let url: URL? = request.mainDocumentURL
+    addressField.text = url?.absoluteString
+    let userData = UserDefaults.standard
+    userData.set(url?.absoluteString, forKey: "KMWebBrowserLastURL")
+    userData.synchronize()
+  }
+
+  func refresh(_ sender: Any) {
+    webView.reload()
+  }
+
+  func stop(_ sender: Any?) {
+    webView.stopLoading()
+  }
+
+  @IBAction func back(_ sender: Any) {
+    webView.goBack()
+  }
+
+  @IBAction func forward(_ sender: Any) {
+    webView.goForward()
+  }
+
+  @IBAction func bookmarks(_ sender: Any) {
+    let bookmarksVC = BookmarksViewController()
+    bookmarksVC.webBrowser = self
+    present(bookmarksVC, animated: true, completion: nil)
+  }
+
+  @IBAction func globe(_ sender: Any) {
+    KMManager.sharedInstance().showKeyboardPicker(in: self, shouldAddKeyboard: false)
+  }
+
+  func close(_ sender: Any?) {
+    dismiss(animated: true, completion: nil)
+  }
+
+  // MARK: - UIWebViewDelegate methods
+  func webView(_ webView: UIWebView,
+               shouldStartLoadWith request: URLRequest,
+               navigationType: UIWebViewNavigationType) -> Bool {
+    updateAddress(request)
+    return true
+  }
+
+  func webViewDidStartLoad(_ webView: UIWebView) {
+    UIApplication.shared.isNetworkActivityIndicatorVisible = true
+    updateButtons()
+  }
+
+  func webViewDidFinishLoad(_ webView: UIWebView) {
+    UIApplication.shared.isNetworkActivityIndicatorVisible = false
+    appendCSSFontFamily()
+    updateButtons()
+  }
+
+  func webView(_ webView: UIWebView, didFailLoadWithError error: Error) {
+    UIApplication.shared.isNetworkActivityIndicatorVisible = false
+    updateButtons()
+    let alert = UIAlertView(title: "Cannot Open Page",
```

Comment:
The old implementation conditioned on two error codes: -1003 and -1009.  Is there a reason we're changing this behavior?  I mean, I could see it being reasonable, but just wanting to ensure it's been vetted properly.

Reply:
References to errors:
https://developer.apple.com/documentation/foundation/1508628-url_loading_system_error_codes/nsurlerrorcannotfindhost?language=objc
https://developer.apple.com/documentation/foundation/1508628-url_loading_system_error_codes/nsurlerrornotconnectedtointernet?language=objc

So `NSURLErrorCannotFindHost = -1003` and `NSURLErrorNotConnectedToInternet = -1009`. The original reason is that it's a bit awkward to access the error code as `(error as NSError).code`. Looking at the other errors though to dig up what those magic numbers were, it seemed like we should show an alert from them as well rather than silently swallowing it.

These are all the other errors:
https://developer.apple.com/documentation/foundation/1448136-nserror_codes?language=objc